server/sonar-ce/src/main/java/org/sonar/ce/taskprocessor/CeProcessingSchedulerImpl.java

```diff
@@ -87,7 +87,7 @@ public void stopScheduling() {
       try {
         Thread.sleep(200L);
       } catch (InterruptedException e) {
-        LOG.debug("Graceful stop period has been interrupted", e);
+        LOG.debug("Graceful stop period has been interrupted: {}", e);
```

Comment:
good catch !

Reply:
Are you sure? Here the point is to attache the stacktrace to the log. Not to log is as part of the message.

Reply:
I agree, that's not a good catch at all. The stacktrace of the error is lost.

Reply:
`LOG` here is not an SLF4J/Logback logger, but a `org.sonar.api.utils.log.Logger`:

https://github.com/SonarSource/sonarqube/blob/6bf0c30c664b7eb6e817b74d0a5e2948627c0caa/server/sonar-ce/src/main/java/org/sonar/ce/taskprocessor/CeProcessingSchedulerImpl.java#L38

If you take a look at the [`org.sonar.api.utils.log.Logger` interface](https://github.com/SonarSource/sonarqube/blob/6bf0c30c664b7eb6e817b74d0a5e2948627c0caa/sonar-plugin-api/src/main/java/org/sonar/api/utils/log/Logger.java), only at the warn and error levels is there the ability to log an exception/throwable with full stack trace.

With regard to this specific change, the absence of a debug(String, Throwable) method means that debug(String, @Nullable Object) is called instead. And because there is no formatting anchor `{}` in the format string, no information about the exception would be logged; the only thing that would be logged is "Graceful stop period has been interrupted".

If preserving the stack trace is desired, then the log should be at the warn or error level.